show_client/ndp_cli.go

```diff
@@ -0,0 +1,418 @@
+					break
+				}
+			}
+		}
+
+		// Get Status (last field)
+		status = fields[len(fields)-1]
+
+		entry := NeighborEntry{
+			Address:    address,
+			MacAddress: mac,
+			Iface:      iface,
+			Vlan:       vlan,
+			Status:     status,
+		}
+
+		table.Entries = append(table.Entries, entry)
+	}
+
+	// Sort table by IP address (like natsorted in Python)
+	sort.Slice(table.Entries, func(i, j int) bool {
+		return table.Entries[i].Address < table.Entries[j].Address
+	})
+
+	table.TotalEntries = len(table.Entries)
+	return table
+}
+
+func getInterfaceOidMap() (map[string]string, error) {
+	portQueries := [][]string{
+		{"COUNTERS_DB", "COUNTERS_PORT_NAME_MAP"},
+	}
+	lagQueries := [][]string{
+		{"COUNTERS_DB", "COUNTERS_LAG_NAME_MAP"},
+	}
+
+	portMap, err := GetMapFromQueries(portQueries)
+	if err != nil {
+		return nil, err
+	}
+	lagMap, err := GetMapFromQueries(lagQueries)
+	if err != nil {
+		return nil, err
+	}
+
+	// SONiC interface regex patterns
+	ethRe := regexp.MustCompile(`^Ethernet(\d+)$`)
+	lagRe := regexp.MustCompile(`^PortChannel(\d+)$`)
+	vlanRe := regexp.MustCompile(`^Vlan(\d+)$`)
+	mgmtRe := regexp.MustCompile(`^eth(\d+)$`)
+
+	oidPrefix := len("oid:0x")
+	ifOidMap := make(map[string]string)
+
+	// helper closure to check valid names
+	isValidIfName := func(name string) bool {
+		return ethRe.MatchString(name) ||
+			lagRe.MatchString(name) ||
+			vlanRe.MatchString(name) ||
+			mgmtRe.MatchString(name)
+	}
+
+	for portName, oidVal := range portMap {
+		oidStr, ok := oidVal.(string)
+		if !ok {
+			continue
+		}
+		if isValidIfName(portName) {
+			ifOidMap[oidStr[oidPrefix:]] = portName
+		}
+	}
+	for lagName, oidVal := range lagMap {
+		oidStr, ok := oidVal.(string)
+		if !ok {
+			continue
+		}
+		if isValidIfName(lagName) {
+			ifOidMap[oidStr[oidPrefix:]] = lagName
+		}
+	}
+
+	return ifOidMap, nil
+}
+
+func buildBvidToVlanMap() (map[string]string, error) {
+	queries := [][]string{
+		{"ASIC_DB", "ASIC_STATE:SAI_OBJECT_TYPE_VLAN:*"},
+	}
+
+	vlanData, err := GetMapFromQueries(queries)
+	if err != nil {
+		return nil, err
+	}
+
+	const prefix = "SAI_OBJECT_TYPE_VLAN:"
+	result := make(map[string]string)
+
+	for key, val := range vlanData {
+		if !strings.HasPrefix(key, prefix) {
+			continue
+		}
+
+		bvid := strings.TrimPrefix(key, prefix) // "oid:..."
+
+		ent, ok := val.(map[string]interface{})
+		if !ok {
+			log.Warningf("Unexpected format for VLAN entry %s: %#v", key, val)
+			continue
+		}
+
+		if vlanIDRaw, ok := ent["SAI_VLAN_ATTR_VLAN_ID"]; ok {
+			if vlanIDStr, ok := vlanIDRaw.(string); ok {
+				result[bvid] = vlanIDStr
+			}
+		}
+	}
+
+	return result, nil
+}
+
+func getVlanIDFromBvid(bvid string, bvidMap map[string]string) (string, error) {
+	if vlanID, ok := bvidMap[bvid]; ok {
+		return vlanID, nil
+	}
+	return "", fmt.Errorf("BVID %s not found in VLAN map", bvid)
+}
+
+func getBridgePortMap() (map[string]string, error) {
+	queries := [][]string{
+		{"ASIC_DB", "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:*"},
```

Comment:
Do we need wildcard at the end can you doublecheck?

Reply:
double checked, we need this wildcard otherwise will return emtpy